### PR TITLE
Update reduction_ops groupings to include primtorch types

### LINKED ISSUE #107335: Update generic operator tests to run for primtorch ops
## Issue description

Currently most of the operator groupings in [common_method_invocations](https://github.com/pytorch/pytorch/blob/8507b22fea9ef819452655aa6dd1c42752a4e74a/torch/testing/_internal/common_methods_invocations.py#L20638-L20651) don't include the reference operators and this could lead to unintended mismatch in operator functionality between eager ops and primtorch ops slipping through. Some cases have already been seen of this while creating https://github.com/pytorch/pytorch/pull/104489.  This is issue is being created in order to track and breakdown the changes to multiple pull requests for more manageable pieces for review. The `sparse_*` groupings cannot be currently updated as it seems that sparse tensor support for primtorch is is not yet complete.

https://github.com/pytorch/pytorch/blob/8507b22fea9ef819452655aa6dd1c42752a4e74a/torch/testing/_internal/common_methods_invocations.py#L20638-L20651
- [ ] Update `unary_ufuncs` groupings
- [ ] Update `binary_ufuncs` groupings
- [ ] Update `spectral_funcs` groupings
- [ ] Update `sparse_unary_ufuncs` groupings
- [ ] Update `sparse_csr_unary_ufuncs` groupings
- [ ] Update `sparse_reduction_ops` groupings
- [ ] Update `shape_funcs` groupings
- [ ] Update `reduction_ops` groupings

cc @ezyang @mruberry @Lezcano @peterbell10

### PR DESCRIPTION
**Pull request by ekamiti**
Fixes https://github.com/pytorch/pytorch/issues/107335. The skips were updated for the _ref ops to match those for eager mode where necessary. Part of breakdown of https://github.com/pytorch/pytorch/pull/104489.

### CODE REVIEW

File: test/test_reductions.py [hunk 1]
```diff
@@ -296,8 +296,13 @@ def test_empty_tensor_empty_slice(self, device, op: ReductionOpInfo):
                 self.assertEqual(result, torch.full_like(result, torch.nan))
             else:
                 # Reducing along empty slice should raise an error
-                with self.assertRaises(IndexError):
-                    op(t, *args, dim=dim, **kwargs)
+                if isinstance(op, ReductionPythonRefInfo):
+                    # ref reductions throw RuntimeError for this
+                    with self.assertRaises(RuntimeError):
```
Comment (ezyang) [inline test/test_reductions.py:301]: If possible, we should fix the refs here. Won't block this PR on it though

**Review by ezyang — APPROVED** [review]

File: test/test_reductions.py (hunk 1 above)
Comment (ekamiti) [reply test/test_reductions.py:301]: Filed issue https://github.com/pytorch/pytorch/issues/107494 to track this.

**Pull request closed without merging**